cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/TransactionServiceManager.java

```diff
@@ -35,14 +35,16 @@
  */
 public class TransactionServiceManager extends AbstractDistributedMasterServiceManager {
   private static final Logger LOG = LoggerFactory.getLogger(TransactionServiceManager.class);
-  private TransactionSystemClient txClient;
-  private DiscoveryServiceClient discoveryServiceClient;
+  private final TransactionSystemClient txClient;
+  private final DiscoveryServiceClient discoveryServiceClient;
+  private final boolean isSql;
 
   @Inject
   public TransactionServiceManager(CConfiguration cConf, TwillRunnerService twillRunnerService,
                                    TransactionSystemClient txClient, DiscoveryServiceClient discoveryServiceClient) {
     super(cConf, Constants.Service.TRANSACTION, twillRunnerService, discoveryServiceClient);
     this.txClient = txClient;
+    this.isSql = cConf.get(Constants.Dataset.DATA_STORAGE_IMPLEMENTATION).equals(Constants.Dataset.DATA_STORAGE_SQL);
```

Comment:
I feel that going forward, it should still be possible to use SQL and also Tephra transactions. Also, if we allow separate configurations for the Table SPI and Metadata SPI, then it is possible that the Metadata is still in datasets (and I believe it always is like that in preview) and might need the transaction server, despite the SQL binding for tables. 

Can we make this a separate configuration in cdap-site.xml, whether to start the tx service or not?

Reply:
I added an option to disable tx service. And I checked for preview, preview does not record any system metadata, lineage, and data accessor. I can run preview successfully after I turn off tx service

Reply:
That is because preview itself does not interact with metadata. I wonder what happens if a plugin uses the context to write metadata. Preview does inject the Dataset-based metadata storage.  

See https://github.com/cdapio/cdap/blob/develop/cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/AbstractStageContext.java#L185 and https://github.com/cdapio/cdap/blob/develop/cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/preview/PreviewDataModules.java#L73

Reply:
Perhaps we should inject NoopMetadataStorage instead

Reply:
Updated to NoopMetadataStorage, please take another look, thank you!